src/silo/storage/metadata_store.cpp

```diff
@@ -43,28 +41,39 @@ unsigned MetadataStore::fill(
    return sequence_count;
 }
 
-void MetadataStore::initializeColumns(
-   const config::DatabaseConfig& database_config,
-   const std::set<std::string>& columns_to_index
-) {
+void MetadataStore::initializeColumns(const config::DatabaseConfig& database_config) {
    for (const auto& item : database_config.schema.metadata) {
       if (item.type == config::DatabaseMetadataType::STRING) {
-         if (columns_to_index.contains(item.name)) {
-            this->indexed_string_columns[item.name] = storage::column::IndexedStringColumn();
-         } else {
-            this->raw_string_columns[item.name] = storage::column::RawStringColumn();
-         }
+         raw_string_columns[item.name] = storage::column::RawStringColumn();
+      } else if (item.type == config::DatabaseMetadataType::INDEXED_STRING) {
+         indexed_string_columns[item.name] = storage::column::IndexedStringColumn();
       } else if (item.type == config::DatabaseMetadataType::PANGOLINEAGE) {
          pango_lineage_columns.emplace(item.name, storage::column::PangoLineageColumn());
       } else if (item.type == config::DatabaseMetadataType::DATE) {
          date_columns.emplace(
             item.name,
-            // TODO(#114) make this configurable
-            item.name == "date" ? storage::column::DateColumn(true)
-                                : storage::column::DateColumn(false)
+            item.name == database_config.schema.date_to_sort_by ? storage::column::DateColumn(true)
+                                                                : storage::column::DateColumn(false)
          );
       }
    }
 }
 
+const storage::column::Column& MetadataStore::getColumn(const config::DatabaseMetadata& metadata
```

Comment:
Why is this unused? Is it preparation for something else?

Reply:
Yes, it is used by the code for the aggregation endpoint